plugins/couchbase/CMakeLists.txt

```diff
@@ -25,37 +25,68 @@
 project(couchbaseembed)
 
 IF ( COUCHBASEEMBED )
-    ADD_PLUGIN(COUCHBASEEMBED PACKAGES COUCHBASE)
-    IF ( MAKE_COUCHBASEEMBED )
-        set(
-            SRCS
-            couchbaseembed.cpp
+
+    if(APPLE)
+        set(LIBCOUCHBASE_LIB ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so)
+        set(LIBCOUCHBASE_LIB_SONAME ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so.2)
+        set(LIBCOUCHBASE_LIB_REAL ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so.2.0.38)
+    else()
+        set(LIBCOUCHBASE_LIB ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so)
+        set(LIBCOUCHBASE_LIB_SONAME ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so.2)
+        set(LIBCOUCHBASE_LIB_REAL ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build/lib/libcouchbase.so.2.0.38)
+    endif()
+    add_custom_command(
+        OUTPUT ${LIBCOUCHBASE_LIB}
+        COMMAND mkdir -p ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build
+        COMMAND cd ${CMAKE_CURRENT_BINARY_DIR}/libcouchbase-build && ${CMAKE_COMMAND} ${CMAKE_CURRENT_SOURCE_DIR}/libcouchbase && ${CMAKE_MAKE_PROGRAM}
```

Comment:
${CMAKE_COMMAND} and ${CMAKE_MAKE_PROGRAM} fix an issue where cmake can't be found in the shell environment that is on the smoketest